lightning/src/ln/channelmanager.rs

```diff
@@ -6985,6 +7128,14 @@ impl<'a, Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 			};
 		}
 
+		let pending_intercepted_payment_count: u64 = Readable::read(reader)?;
+		let mut pending_intercepted_payments: HashMap<InterceptId, PendingInterceptedHTLC> = HashMap::with_capacity(cmp::min(pending_intercepted_payment_count as usize, MAX_ALLOC_SIZE/(3*32)));
+		for _ in 0..pending_intercepted_payment_count {
+			if pending_intercepted_payments.insert(Readable::read(reader)?, Readable::read(reader)?).is_some() {
+				return Err(DecodeError::InvalidValue);
+			}
+		}
+
```

Comment:
tried to read it last as well but the reader var gets moved into the read_tlv_fields! for outbound_payments below.  just put it here for now so it compiles but I'm pretty sure this won't work.